### PR TITLE
3844 add front end UI pop data entry

### LINKED ISSUE #3844: GAPS: Frontend Code to manage population data entry
Part of epic #3663 

## Is your feature request related to a problem? Please describe 👀
| User story  | B.2.11 |
| :------------- | :------------- |
As a… | National EPI manager
I want… | all facilities to provide the central EPI office with the number of people they serve in their catchment area
So that… | the central EPI office can leverage the facilities' localized knowledge to build a picture of the demographics that the immunization supply chain is catering for nationally

**Solution B.2.11**
The system will provide a mechanism for facilities to enter a figure to represent the number of people they serve. 
For facilities that are not using Open mSupply central users will be able to enter the figure on their behalf. 
All figures will be aggregated at the central level and the aggregated total will serve as the ‘General population’ figure used by the demographic groups detailed in User Story B.2.2.  **_<-- after a discussion with @jmbrunskill decided that aggregation is not actually useful except for cross-referencing purposes and so we should allow central user to specify their own `General Population` value_**

---
| User story  | B.2.14 |
| :------------- | :------------- |
As a… | National EPI manager and National PHC manager
I want… | to be able to specify the expected population growth on an annual basis, for the next 5 years
So that… | the system can show me the change in cold storage situation that this growth causes for each of the next 5 years

**Solution B.3.1**
The system will allow central users to enter a percentage value to represent estimated population growth for each of the following 5 years. 
This value will be used in conjunction with the current general population aggregate total to arrive at future estimated general population total. 
This will then make it possible to estimate the national cold storage capacity demand for each of the following 5 years. 

---

UI to manage the population and population growth for a country as part of the Gaps analysis, we need a ui which dynamically updates as population, and growth rates are updated.

For this issue, we'll simply provide the UI to enter the population data, and growth rates.
A subsequent issues will add the ability to define new indicators, and save the data.

### Acceptance Criteria

- [ ] A new page should be added to the central server navigation called` Indicators Demographics`
- [ ] The new page should only be available from the central server
- [ ] The screen should open with the current year + 5 years shown along the top
- [ ] User can enter a general population figure (This will be temporary as in future it will be automatically aggregated)
- [ ] User can enter a growth rate for each of the five years shown
- [ ] Entering a new growth rate, causes the estimated future populations to update on the screen

### Additional context 💌

Part of https://github.com/msupply-foundation/open-msupply/issues/3662 and https://github.com/msupply-foundation/open-msupply/issues/3663

### Moneyworks Jobcode 🧰

CCEIDEV

### PR DESCRIPTION
**Pull request by fergie-nz**
Fixes #3844

# 👩🏻‍💻 What does this PR do?

Adds front end demographic dashboard.

## 💌 Any notes for the reviewer?

You can view this currently via the catalogue nav on central server (this will later be moved to a new menu item)

# 🧪 Testing

# 📃 Documentation

### CODE REVIEW

File: client/packages/common/src/ui/layout/tables/components/Header/Header.tsx [hunk 1]
```diff
@@ -63,7 +63,11 @@ export const HeaderCell = <T extends RecordWithId>({
   );
 
   const columnLabel =
-    column.label === '' ? '' : t(column.label, column.labelProps);
+    column.label === ''
```
Comment (roxy-dao) [inline client/packages/common/src/ui/layout/tables/components/Header/Header.tsx:66]: Can we refactor this so it's not a nested ternary 🙏🏻

File: client/packages/common/src/ui/layout/tables/components/index.tsx [hunk 2]
```diff
@@ -45,6 +45,11 @@ export const BasicHeader = <T extends RecordWithId>({
   column,
 }: HeaderProps<T>): ReactElement => {
   const t = useTranslation();
-  const header = column.label === '' ? '' : t(column.label, column.labelProps);
+  const header =
```
Comment (roxy-dao) [inline client/packages/common/src/ui/layout/tables/components/index.tsx:48]: Same as above 🙏🏻

File: client/packages/host/src/components/Navigation/CatalogueNav.tsx [hunk 3]
```diff
@@ -49,6 +52,17 @@ export const CatalogueNav: FC = () => {
               .build()}
             text={t('master-lists')}
           />
+          <AppNavLink
+            visible={
+              // isCentralServer && Environment.FEATURE_DEMOGRAPHIC_INDICATORS
```
Comment (roxy-dao) [inline client/packages/host/src/components/Navigation/CatalogueNav.tsx:57]: Are we not hiding this behind a feature flag for now~?

**Review by roxy-dao — COMMENTED** [review]
Hai~ I just had a brief look at the first few files (since I don't have enough context XD)

**Review by mark-prins — COMMENTED** [review]
Looking good thanks Ferg! 
Just a few commented lines to tidy and a couple of suggestions 🙏

File: client/packages/system/src/IndicatorsDemographics/DetailView/PercentageColumn.tsx [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+>(): ColumnDefinition<T> => {
+  return {
+    label: 'label.percentage',
+    setter: () => {
+      if (process.env['NODE_ENV']) {
+        throw new Error(
+          `The default setter of the NameAndColor column was called.
+        Have you forgotten to provide a custom setter?
+        When setting up your columns, you should provide a setter function
+        const columns = useColumns([ gercentageColumn(), { setter }])
```
Comment (mark-prins) [inline client/packages/system/src/IndicatorsDemographics/DetailView/PercentageColumn.tsx:22]:
```suggestion
        const columns = useColumns([ percentageColumn(), { setter }])
```

File: client/packages/system/src/IndicatorsDemographics/DetailView/PercentageColumn.tsx [hunk 5]
```diff
@@ -0,0 +1,56 @@
...
+        sx={{
+          flexDirection: 'row',
+          borderBottom: 'none',
+          alignItems: 'center',
+          display: 'flex',
+        }}
+      >
+        <>
+          <BasicTextInput
+            disabled={isDisabled || rowData.name == 'General Population'}
```
Comment (mark-prins) [inline client/packages/system/src/IndicatorsDemographics/DetailView/PercentageColumn.tsx:40]: Using rowData.id === `1` would be safer. the 'gen pop' will need translation-ability

Comment (mark-prins) [reply client/packages/system/src/IndicatorsDemographics/DetailView/PercentageColumn.tsx:40]: Oh, and using `==` doesn't test for strict equality. best to use `===` if you can

File: client/packages/system/src/IndicatorsDemographics/DetailView/PopulationColumn.tsx [hunk 6]
```diff
@@ -0,0 +1,56 @@
...
+        sx={{
+          flexDirection: 'row',
+          borderBottom: 'none',
+          alignItems: 'center',
+          display: 'flex',
+        }}
+      >
+        <>
+          <BasicTextInput
+            disabled={isDisabled || rowData.name != 'General Population'}
```
Comment (mark-prins) [inline client/packages/system/src/IndicatorsDemographics/DetailView/PopulationColumn.tsx:40]: maybe `rowData.id !== 1`

File: client/packages/system/src/IndicatorsDemographics/DetailView/NameColumn.tsx [hunk 7]
```diff
@@ -0,0 +1,54 @@
...
+        sx={{
+          flexDirection: 'row',
+          borderBottom: 'none',
+          alignItems: 'center',
+          display: 'flex',
+        }}
+      >
+        <>
+          <BasicTextInput
+            disabled={isDisabled || rowData.name == 'General Population'}
```
Comment (mark-prins) [inline client/packages/system/src/IndicatorsDemographics/DetailView/NameColumn.tsx:42]: as noted
- `===`
- `rowData.id === 1`

File: client/packages/system/src/IndicatorsDemographics/DetailView/PopulationColumn.tsx [hunk 8]
```diff
@@ -0,0 +1,56 @@
...
+              });
+            }}
+          />
+          <Box ml={1} />
+        </>
+      </Box>
+    ),
+    minWidth: 100,
+  };
+};
```
Comment (mark-prins) [inline client/packages/system/src/IndicatorsDemographics/DetailView/PopulationColumn.tsx:56]: no need to make a new function here 
```suggestion
>(): ColumnDefinition<T> => ({
  label: 'label.population',
  setter: () => {
    if (process.env['NODE_ENV']) {
      throw new Error(
        `The default setter of the NameAndColor column was called.
        Have you forgotten to provide a custom setter?
        When setting up your columns, you should provide a setter function
        const columns = useColumns([ gercentageColumn(), { setter }])
        `
      );
    }
  },
  accessor: ({ rowData }) => rowData[0],
  key: '0',
  Cell: ({ rowData, column, isDisabled }) => (
    <Box
      sx={{
        flexDirection: 'row',
        borderBottom: 'none',
        alignItems: 'center',
        display: 'flex',
      }}
    >
      <>
        <BasicTextInput
          disabled={isDisabled || rowData.name != 'General Population'}
          defaultValue={column.accessor({ rowData }) as number}
          onBlur={e => {
            const updatedRowData = { ...rowData };
            column.setter({
              ...updatedRowData,
              0: parseInt(e.target.value),
            });
          }}
        />
        <Box ml={1} />
      </>
    </Box>
  ),
  minWidth: 100,
});
```

File: client/packages/system/src/IndicatorsDemographics/DetailView/NameColumn.tsx [hunk 9]
```diff
@@ -0,0 +1,54 @@
...
+              column.setter({ ...rowData, name: e.target.value });
+            }}
+          />
+          <Box ml={1} />
+        </>
+      </Box>
+    ),
+    minWidth: 200,
+  };
+};
```
Comment (mark-prins) [inline client/packages/system/src/IndicatorsDemographics/DetailView/NameColumn.tsx:54]:
```suggestion
>(): ColumnDefinition<T> => ({
  label: 'label.name',
  setter: () => {
    if (process.env['NODE_ENV']) {
      throw new Error(
        `The default setter of the NameAndColor column was called.
        Have you forgotten to provide a custom setter?
        When setting up your columns, you should provide a setter function
        const columns = useColumns([ percentageColumn(), { setter }])
        `
      );
    }
  },
  accessor: ({ rowData }) => rowData.name,
  key: 'name',
  Cell: ({ rowData, column, isDisabled }) => (
    <Box
      sx={{
        flexDirection: 'row',
        borderBottom: 'none',
        alignItems: 'center',
        display: 'flex',
      }}
    >
      <>
        <BasicTextInput
          disabled={isDisabled || rowData.name == 'General Population'}
          defaultValue={column.accessor({ rowData })}
          onBlur={e => {
            column.setter({ ...rowData, name: e.target.value });
          }}
        />
        <Box ml={1} />
      </>
    </Box>
  ),
  minWidth: 200,
});
```

**Review by mark-prins — COMMENTED** [review]
a few more suggestions after looking through

**Review by mark-prins — APPROVED** [review]
Thanks Ferg! I think merge it in, and we can look at the base population thing separately
🚢

**Pull request merged**